### PR TITLE
Add Invisible reCaptcha on Registration Page

### LINKED ISSUE #7674: Invisible reCaptcha

### LINKED ISSUE #7695: Use Invisible ReCaptcha
Minimal design effort needed: simply remove the currently visible reCaptcha box.

Engineers to investigate into the possibility of using the latest version of reCaptcha, which no longer utilizes a visible checkbox box reading "I'm not a robot" before the user could click on the button to view and complete the graph quiz window and then proceed to the next step. Instead, the newest reCaptcha generates the graph quiz window automatically after the user clicks on the button to proceed to the next step.

Belongs to the epic below:
#7694

### PR DESCRIPTION
**Pull request by imlakshay08**
Closes #7674
Closes #7695 

This PR adds a Invisible Google reCaptcha on registration page of Open Library.

### Screenshot
Screenshot_20230319_140409

### Stakeholders

@mekarpeles @cdrini

### CODE REVIEW

**Review by jimchamp — CHANGES_REQUESTED** [review]
Thanks @imlakshay08, and sorry for the delay!
I've added a number of suggested changes that will be needed for this PR.  In addition to those changes, some changes to the reCAPTCHA configuration for the sign-up form are needed:

1. The Open Library staff will need to generate new keys in order to use the invisible reCAPTCHAs.  Can you create a new entry in our `openlibrary.yml` configuration file for these keys?  You can reference the existing keys [here](https://github.com/internetarchive/openlibrary/blob/bee11c445c5d2d37f6d91c56b89791bb138e80cd/conf/openlibrary.yml#L148-L150), and the new entry can be called `invisible_recaptcha`.
2. The keys for the sign up page are set [here](https://github.com/internetarchive/openlibrary/blob/bee11c445c5d2d37f6d91c56b89791bb138e80cd/openlibrary/plugins/upstream/account.py#L258-L259).  This will need to be updated to use the new config entry.

**Comment by imlakshay08** [comment]
> Thanks @imlakshay08, and sorry for the delay! I've added a number of suggested changes that will be needed for this PR. In addition to those changes, some changes to the reCAPTCHA configuration for the sign-up form are needed:
> 
> 1. The Open Library staff will need to generate new keys in order to use the invisible reCAPTCHAs.  Can you create a new entry in our `openlibrary.yml` configuration file for these keys?  You can reference the existing keys [here](https://github.com/internetarchive/openlibrary/blob/bee11c445c5d2d37f6d91c56b89791bb138e80cd/conf/openlibrary.yml#L148-L150), and the new entry can be called `plugin_invisible_recaptcha`.
> 2. The keys for the sign up page are set [here](https://github.com/internetarchive/openlibrary/blob/bee11c445c5d2d37f6d91c56b89791bb138e80cd/openlibrary/plugins/upstream/account.py#L258-L259).  This will need to be updated to use the new config entry.

Hey @jimchamp I have updated the pull requested according to the changes requested. Could you please review it?

File: openlibrary/plugins/upstream/account.py [hunk 1]
```diff
@@ -253,9 +253,10 @@ def get_form(self) -> forms.RegisterForm:
 
     def get_recap(self):
         if self.is_plugin_enabled('recaptcha'):
-            public_key = config.plugin_recaptcha.public_key
-            private_key = config.plugin_recaptcha.private_key
-            return recaptcha.Recaptcha(public_key, private_key)
+        public_key = config.plugin_invisible_recaptcha.public_key  # Updated to use the new config entry
+        private_key = config.plugin_invisible_recaptcha.private_key  # Updated to use the new config entry
+        return recaptcha.Recaptcha(public_key, private_key)
+
```
Comment (jimchamp) [inline openlibrary/plugins/upstream/account.py:259]:
```suggestion
            public_key = config.plugin_invisible_recaptcha.public_key
            private_key = config.plugin_invisible_recaptcha.private_key
            return recaptcha.Recaptcha(public_key, private_key)
```
Indentation levels are meaningful in Python.  The `if` block code needs to be indented one level more than the `if` statement.
The comments can also be removed.  We can look at the code history in `git`, so these aren't really useful.

File: openlibrary/templates/account/create.html [hunk 2]
```diff
@@ -56,22 +57,17 @@ <h1>$_("Sign Up")</h1>
...
+        <div class="label smaller lighter">$_("If you have security settings or privacy blockers installed, please disable them to see the reCAPTCHA.")</div>
+    <br/>
         <small>$:_("By signing up, you agree to the Internet Archive's <a href='//archive.org/about/terms.php' target='_blank'>Terms of Service</a>.")</small>
-
         <div class="formElement bottom">
+            <script>
+                function onSubmit(token) {
+                  document.getElementById("demo-form").submit();
+                }
+              </script>
```
[the remark this answers is not in the document]
Comment (jimchamp) [reply openlibrary/templates/account/create.html:65]: What did you do to resolve this?  Do you have questions about what is being asked?

File: openlibrary/templates/account/create.html [hunk 3]
```diff
@@ -56,22 +57,17 @@ <h1>$_("Sign Up")</h1>
...
             <br/>
-            <button type="submit" name="signup" id="signup" class="larger">$_("Sign Up")</button>
+            <button type="submit" name="signup" id="signup" class="g-recaptcha" data-sitekey="reCAPTCHA_site_key" data-callback='onSubmit' data-action='submit'>$_("Sign Up")</button>
```
[the remark this answers is not in the document]
Comment (jimchamp) [reply openlibrary/templates/account/create.html]: Change number 3 was not implemented.  Do you have questions about this?

**Review by jimchamp — CHANGES_REQUESTED** [review]
Thanks @imlakshay08!
Looks like some things are still needed:
1. Fix Python indentation errors.
2. Remove `<script>` tag in `create.html`.  What is that script even doing?
3. Update the button's click handler to execute reCAPTCHA test.

**Comment by jimchamp** [comment]
@imlakshay08, is this ready for review?

**Comment by imlakshay08** [comment]
> @imlakshay08, is this ready for review?

Yup, it is ready for the review.

File: conf/openlibrary.yml [hunk 4]
```diff
@@ -151,6 +151,10 @@ plugin_recaptcha:
     public_key: ''
     private_key: ''
 
+plugin_invisible_recaptcha:
+  public_key: ''
+  private_key: ''
```
Comment (jimchamp) [inline conf/openlibrary.yml:156]:
```suggestion
    public_key: ''
    private_key: ''
```
Use consistent indentation amount in this file.

**Review by jimchamp — COMMENTED** [review]
The following needed changes were not made:
- Submit button missing `larger` class
- `data-sitekey` not properly set
- ReCAPTCHA test never called in existing submit button click handler
- Unnecessary V3 data attributes still present

@imlakshay08, I encouraged you to ask questions if you didn't understand what I requested.  No questions were asked, nor were the questions that I asked you answered.

I'm not really sure how to move this PR forward.  Closing for now, but happy to work with you towards a working solution, if you desire.

**Pull request closed without merging**